src/formula/string_utils.cpp

```diff
@@ -356,3 +356,43 @@ std::string vngettext_impl(const char* domain,
 	const std::string msg = utils::interpolate_variables_into_string(orig, &symbols);
 	return msg;
 }
+
+int approximate_string_distance(const std::string &str_1, const std::string &str_2)
```

Comment:
Okay… so is this [Damerau-Levenshtein distance](https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance)? That's cool, adding the transposition test. And I'm kinda surprised we didn't already have a utility like this in the engine…

Although I don't really mind this name, the shorter name `edit_distance` might be a bit nicer.

This is probably out of the scope the PR, but it'd also be nice to expose this in Lua in the `stringx` module.

Reply:
Keep in mind that this is not the full Levenshtein's implementation, it does not do the minimum distance of the string's tails. So I would be reluctant to change the function's name to just `edit_distance` I'd say, at minimum,  `approx_edit_distance` or `edit_distance_approx`.

Reply:
Fair enough, either of those would be fine too, though like I said I don't really mind the current name either.